scalding-macros/src/main/scala/com/twitter/scalding/macros/impl/ordser/ByteBufferOrderedBuf.scala

```diff
@@ -0,0 +1,96 @@
+/*
+ Copyright 2014 Twitter, Inc.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+package com.twitter.scalding.macros.impl.ordser
+
+import scala.language.experimental.macros
+import scala.reflect.macros.Context
+
+import com.twitter.scalding._
+import java.nio.ByteBuffer
+import com.twitter.scalding.serialization.OrderedSerialization
+
+object ByteBufferOrderedBuf {
```

Comment:
why does this have to be a Macro? I'm still not seeing the argument as to why basic types should not be implemented with an instance. If we need to use them in other macros get them implicitly or wrap them as you do in the ScroogeInternal macro.